lib/inc/puppetdb-cli/puppetdb-cli.hpp

```diff
@@ -28,10 +28,30 @@ class LIBPUPPETDB_EXPORT PuppetDBConn  {
     PuppetDBConn();
 
     /**
+     * Construct a PuppetDB connection using only config flags
+     * @param urls string of the urls of PuppetDB.
+     * @param cacert string cacert to use for curl.
+     * @param cert client cert to use for curl.
+     * @param key client private key to use for curl.
+     */
+    PuppetDBConn(const std::string& urls,
```

Comment:
Then we're just wrapping things in a class to pass them to another class, this way you know the api by looking at the function and you don't have to go track down the struct